src/bugsnag_tools/core.clj

```diff
@@ -15,14 +15,38 @@
     path
     (str host path)))
 
+(defn retry
+  "Given a function, f, run it after delay-ms milliseconds have
+  passed."
+  [f delay-ms]
+  (Thread/sleep delay-ms)
+  (f))
+
+(defn rate-limit-middleware
```

Comment:
Mind briefly documenting the purpose and rationale of this rate limiter implementation?